### ISSUE TITLE
parser.c is out of date and fails to build on python 3.7

### ISSUE THREAD

**Issue by eli-schwartz** [benoitc/http-parser#76]
Regenerating it should be sufficient.

Note also the following compiler warnings:

```
http_parser/parser.c: In function ‘__pyx_pw_11http_parser_6parser_10HttpParser_1__init__’:
http_parser/parser.c:3932:3: warning: ‘__pyx_v_parser_type’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   http_parser_init((&__pyx_v_self->_parser), __pyx_v_parser_type);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
http_parser/parser.c:3820:25: note: ‘__pyx_v_parser_type’ was declared here
   enum http_parser_type __pyx_v_parser_type;
                         ^~~~~~~~~~~~~~~~~~~
```

Also use of the "imp" module on python 3.7 results in a DeprecationWarning.

**Comment by ariesdevil** [comment]
@benoitc Can I get you to publish a new version of the package when you can? Thank you!

**Comment by eli-schwartz** [comment]
Fixed by #77 ...
